src/main/java/com/conveyal/gtfs/graphql/fetchers/NestedJDBCFetcher.java

```diff
@@ -75,54 +83,71 @@ public static GraphQLFieldDefinition field (String tableName) {
-                joinValues = joinValuesForJoinField.get(fetcher.parentJoinField);
+                // add nested table join by using a where condition
+                fromTables.add(tableName);
+                whereConditions.add(String.format(
+                    "%s = %s",
+                    namespacedTableFieldName(namespace, lastFetcher.tableName, fetcher.parentJoinField),
+                    namespacedTableFieldName(namespace, fetcher.tableName, fetcher.parentJoinField)
+                ));
+                // check if we have reached the final nested table
                 if (i == jdbcFetchers.length - 1) {
-                    // Apply arguments only to the final fetched table
-                    arguments = environment.getArguments();
-                    LOG.info("{} args: {}", fetcher.tableName, arguments.keySet().toString());
-                }
-            }
-            LOG.info("Join values: {}", joinValues.toString());
-            fetchResults = fetcher.getResults(namespace, joinValues, arguments);
-            if (fetchResults.size() == 0) {
-                // If there are no results, the following queries will have no results to join to, so we can simply
-                // return the empty list.
-                return fetchResults;
-            } else if (i < jdbcFetchers.length - 1) {
-                // Otherwise, iterate over results from current fetcher to store for next iteration for all but the last
-                // iteration (the last iteration's fetchResults will contain the final results).
-                JDBCFetcher nextFetcher = jdbcFetchers[i + 1];
-                for (Map<String, Object> entity : fetchResults) {
-                    Object joinValue = entity.get(nextFetcher.parentJoinField);
-                    // Store join values in multimap for
-                    if (joinValue != null)
-                        joinValuesForJoinField.put(nextFetcher.parentJoinField, joinValue.toString());
+                    // final nested table reached!
+                    // create a sqlBuilder to select all values from the final table
+                    StringBuilder sqlStatementStringBuilder = new StringBuilder();
+                    sqlStatementStringBuilder.append("select ");
+                    sqlStatementStringBuilder.append(
+                        namespacedTableFieldName(namespace, fetcher.tableName, "*")
+                    );
+                    // Make the query and return the results!
+                    return fetcher.getResults(
+                        getJdbcQueryDataLoaderFromContext(environment),
+                        getDataSourceFromContext(environment),
+                        namespace,
+                        new ArrayList<>(),
+                        graphQLQueryArguemnts,
+                        preparedStatementParameters,
+                        whereConditions,
+                        fromTables,
+                        sqlStatementStringBuilder
```

Comment:
I find it strange that this string builder argument is expected to contain the first clause of the SQL query (e.g., `select *` or `select abc.table.*`). This isn't documented anywhere and I feel like it will just cause confusion. Couldn't we pass something a little more transparent in here?